packages/editor/src/lib/components/Canvas.tsx

```diff
@@ -60,32 +59,29 @@ export const Canvas = track(function Canvas() {
 		[editor]
 	)
 
-	const { context: patternContext, isReady: patternIsReady } = usePattern()
-
 	const events = useCanvasEvents()
 
-	React.useEffect(() => {
-		if (patternIsReady && editor.isSafari) {
-			const htmlElm = rHtmlLayer.current
-
-			if (htmlElm) {
-				// Wait for `patternContext` to be picked up
-				requestAnimationFrame(() => {
-					htmlElm.style.display = 'none'
-
-					// Wait for 'display = "none"' to take effect
-					requestAnimationFrame(() => {
-						htmlElm.style.display = ''
-					})
-				})
-			}
-		}
-	}, [editor, patternIsReady])
-
 	React.useEffect(() => {
 		rCanvas.current?.focus()
 	}, [])
 
+	const shapeSvgDefs = useValue(
+		'shapeSvgDefs',
+		() => {
+			const shapeSvgDefsByKey = new Map<string, JSX.Element>()
+			for (const util of objectMapValues(editor.shapeUtils)) {
+				if (!util) return
+				const defs = util.getCanvasSvgDefs()
```

Comment:
A minor thing, but I think if we were just returning the component and pushing it into a set, then we wouldn't need the key (could just use the index for the actual component, i.e. key={i}).